shardingsphere-operator/pkg/controllers/shardingsphere_chaos_controller.go

```diff
@@ -330,20 +490,14 @@ func (r *ShardingSphereChaosReconciler) updateJob(ctx context.Context, requireme
 		return err
 	}
 	if exp != nil {
-		if err := r.Delete(ctx, cur); err != nil {
-			return err
-		}
-		if err := ctrl.SetControllerReference(chao, exp, r.Scheme); err != nil {
-			return err
-		}
-		if err := r.Create(ctx, exp); err != nil {
+		if err := r.Delete(ctx, cur); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
i saw that `reconcile.UpdateJob` will return nil, nil if job no need to update, it is as your expect? the name of `UpdateJob` makes me confused.